File: 3Sum Closest/3Sum_Closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        int n = nums.size();
        int diff = 1e9,ans = 0;
        sort(nums.begin(),nums.end());
        for(int i =0;i<n; i++) {
            int j =i+1,k = n-1;
            while(j<k) {
                int sum = nums[i]+nums[j]+nums[k];
                if(abs(sum-target) < diff) {
                    ans = sum;
                    diff = abs(sum-target);
                }
                if(sum > target) k--;
                else j++;
            }
        }
        return ans;
    }
};

File: 3Sum Closest/3Sum_Closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        n = len(nums)
        diff = float('inf')
        ans = 0
        nums.sort()
        for i in range(n):
            j, k = i + 1, n - 1
            while j < k:
                sum_ = nums[i] + nums[j] + nums[k]
                if abs(sum_ - target) < diff:
                    ans = sum_
                    diff = abs(sum_ - target)
                if sum_ > target:
                    k -= 1
                else:
                    j += 1
        return ans
        